http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/SslCloseCompletionEventHandler.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.nio.netty.internal;
+
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.handler.ssl.SslCloseCompletionEvent;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+
+/**
+ * Handlers {@link SslCloseCompletionEvent}s that can are sent whenever an SSL channel
+ * goes inactive. This most commonly occurs on a tls_close sent by the server. Channels
+ * in this state can't be reused so we must close them.
+ */
+@SdkInternalApi
+public class SslCloseCompletionEventHandler extends ChannelInboundHandlerAdapter {
+
+    @Override
+    public void userEventTriggered(ChannelHandlerContext ctx, Object evt) {
+        if (evt instanceof SslCloseCompletionEvent) {
+            ctx.close();
+        }
+    }
```

Comment:
Should this fix be pushed upstream to netty?

Reply:
As far as I know, doing what we are doing here is standard practice. I'm not sure why Netty doesn't close the channel the SslHandler.